doc/API.md

```diff
@@ -28,17 +32,379 @@ struct {
+
+Parameters:
+```
+struct {
+	encryptionKey string
+	dictionary    string
+}
+```
+'verification' must be set to true, or an error is returned. This is to prevent
+an implementation mistake where POST is called instead of GET or PUT.
+
+'dictionary' is the name of the dictionary that should be used when encoding
+the newly created seed. 'english' is the most common choice.
+
+Response:
+```
+struct {
+	newSeed string
+}
+```
+'newSeed' is the new seed that will be used as the seed for generating new
+addresses.
+
+#### /wallet/siacoins [PUT]
+
+Function: Send siacoins to an address. The outputs are arbitrarily selected
+from addresses in the wallet.
+
+Parameters:
+```
+struct {
+	amount      int
+	destination string
+}
+```
+'amount' is the number of siacoins being sent.
+
+'destination' is the address that is receiving the coins.
+
+Response: standard
+
+#### /wallet/siafunds [PUT]
+
+Function: Send siafunds to an address. The outputs are arbitrarily selected
+from addresses in the wallet. Any siacoins available in the siafunds being sent
+(as well as the siacoins available in any siafunds that end up in a refund
+address) will become available to the wallet as siacoins after 144
+confirmations. To access all of the siacoins in the siacoin claim balance, send
+all of the siafunds to an address in your control (this will give you all the
+siacoins, while still letting you control the siafunds).
+
+Parameters:
+```
+struct {
+	amount      int
+	destination string
+}
+```
+'amount' is the number of siafunds being sent.
+
+'destination' is the address that is receiving the funds.
+
+Response: standard
+
+#### /wallet/transaction/$(id) [GET]
+
+Function: Get the transaction associated with a specific transaction id.
 
 Parameters: none
 
 Response:
 ```
-UNDECIDED
+struct {
+	transaction Transaction
+}
+```
+'transaction' is a 'types.Transaction'. The full transaction can be seen in
+types.transaction.go. All hashes in the transaction are encoded as strings.
+
+#### /wallet/transactions [GET]
+
+Function: Return all raw transactions relevant to the wallet in a block range.
+Raw transactions are missing metadata such as confirmation height.
+
+Parameters:
 ```
+struct {
+	startHeight int
+	endHeight   int
+}
+```
+'startHeight' refers to the height of the block where transaction history
+should begin.
+
+'endHeight' refers to the height of of the block where the transaction history
+should end. If 'endHeight' is greater than the current height, all transactions
+up to and including the most recent block will be provided.
+
+Response:
+```
+struct {
+	confirmedTransactions   []types.Transaction
+	unconfirmedTransactions []types.Transaction
+}
+```
+'confirmedTransactions' lists all of the confirmed transactions appearing
+between height 'start' and height 'end' (inclusive).
+
+'unconfirmedTransactions' lists all of the unconfirmed transactions.
+
+#### /wallet/unlock [PUT]
```

Comment:
And also for /wallet/close?

/wallet/unlock and /wallet/close are inverses of eachother. Also, you can cycle between them repeatedly.